src/commands/deployment/execute.rs

```diff
@@ -716,3 +815,133 @@ async fn get_deployment_status(
+                .map(|build| {
+                    let commits: Vec<Commit> = build
+                        .commits
+                        .into_iter()
+                        .map(|commit| Commit {
+                            id: commit.id,
+                            author: commit.author,
+                            message_headline: commit.message_headline,
+                        })
+                        .collect();
+
+                    JenkinsBuild {
+                        build_duration: build.build_duration,
+                        build_number: build.build_number,
+                        build_branch: build.build_branch,
+                        build_url: build.build_url,
+                        build_artifact_name: build.build_artifact_name,
+                        name: build.name,
+                        result: build.result,
+                        timestamp: build.timestamp,
+                        total_duration: build.total_duration,
+                        wait_duration: build.wait_duration,
+                        commits,
+                    }
+                })
+                .collect();
+
+            let cd_pipeline_with_builds = CdPipelineWithBuilds {
+                name: jenkins_cd_pipeline.name,
+                version: jenkins_cd_pipeline.version,
+                enabled: jenkins_cd_pipeline.enabled,
+                deployed_ref: jenkins_cd_pipeline.deployed_ref,
+                build_artifact: jenkins_cd_pipeline.build_artifact,
+                last_successfully_deployed_artifact: jenkins_cd_pipeline
+                    .last_successfully_deployed_artifact,
+                last_deployed_at: jenkins_cd_pipeline.last_deployment,
+                status: jenkins_cd_pipeline.status,
+                github_builds: Vec::new(),
+                jenkins_builds,
+            };
+
+            Ok(Some(cd_pipeline_with_builds))
+        }
+    }
+}
+
+async fn get_github_cd_pipeline(
+    client: &mut QueryClient,
+    application: &str,
+    namespace: &str,
+    version: &str,
+) -> Result<Option<CdPipelineWithBuilds>, CliError> {
+    let github_cd_pipeline = client
+        .fetch_github_cd_pipeline(application, namespace, version)
+        .await?
+        .data
+        .unwrap()
+        .cd_pipeline;
+
+    match github_cd_pipeline {
+        None => Ok(None),
+        Some(github_cd_pipeline) => {
+            let github_builds = github_cd_pipeline
+                .github_builds
+                .into_iter()
+                .map(|build| {
+                    let commits: Vec<Commit> = build
+                        .commits
+                        .into_iter()
+                        .map(|commit| Commit {
+                            id: commit.id,
+                            author: commit.author,
+                            message_headline: commit.message_headline,
+                        })
+                        .collect();
+
+                    JenkinsBuild {
```

Comment:
Perhaps the name should be changed? since now it is no longer only for Jenkins.

Reply:
Renamed `JenkinsBuild` to `CdPipelineBuild`